[docs][joy-ui] Revise the Overview page (#38842)
mui/material-ui: docs/data/joy/getting-started/overview/overview.md

Look for the [`package: joy-ui`](https://github.com/mui/material-ui/labels/package%3A%20joy-ui) label on open issues and pull requests in the `mui/material-ui` repository on GitHub to see what other community members are working on, and feel free to submit your own.
:::

## Why use Joy UI

Maintained by MUI, **Joy UI is an alternative to Material UI** for projects that **aren't planning to adhere to or use the Material Design** guidelines as a design system starting point.

These two sister libraries feature many of the same components, with similarly designed component APIs and customization features.

> Joy UI is built by applying years of learning with Material UI for an audience looking for sleek design, powerful DX, and extensible components.

Author: I like it! 🤙

The [Order Dashboard](/joy-ui/getting-started/templates/order-dashboard/) template below, available on the [Templates](/joy-ui/getting-started/templates/) page, uses very light customizations to demonstrate how carefully designed all components were, ensuring they work well together.

<img src="/static/joy-ui/overview/order-dashboard.png" style="width: 814px; margin-top: 4px; margin-bottom: 8px;" alt="The Order Dashboard template, inspired by Untitled UI and built by the MUI team using Joy UI with very little customizations." width="1628" height="400" />

### Highly customizable

You should feel inspired and empowered to change, extend, and revamp Joy UI's appearance and behavior with ease.
Drawing from many years of experience with Material UI, Joy UI applies new approaches to customization, enabling you to customize every piece of the components to match your unique design.

Using [Gatsby's documentation side nav](https://www.gatsbyjs.com/docs/) as an example, the demo below shows how to customize the [List](/joy-ui/react-list/) component to look like it using built-in CSS variables.

{{"demo": "../../components/list/ExampleCollapsibleList.js"}}

### Developer experience

> One of the main goals of Joy UI is to spark joy in the creative process of building apps.

Reviewer: I think this is more impactful than just saying it's one of the main goals.

```suggestion
Joy UI draws its name from the idea that it should spark joy in the creative process of building apps.
```

> Promoting an unrivaled developer experience is at the heart of this.

Reviewer: ```suggestion
Providing an unrivaled developer experience is at the heart of this.
```

> Take as an example of this the automatic adjustment of the [Input](/joy-ui/react-input/) component, whereby by customizing its border radius, all components inside it will adapt accordingly, ensuring a polished design _and_ saving you time.

Reviewer: ```suggestion
For example, observe how each element of the [Input](/joy-ui/react-input/) component automatically calculates and adjusts its own dimensions relative to the border radius, saving you from the tedium of doing it yourself:
```

{{"demo": "../../main-features/automatic-adjustment/InputVariables.js"}}

### Accessibility

> Joy UI components are built on top of [Base UI's unstyled components and low-level hooks](/base-ui/getting-started/), giving you many accessibility features support out of the box.

Reviewer: ```suggestion
Joy UI components are built on top of [Base UI's unstyled components and low-level hooks](/base-ui/getting-started/), giving you many accessibility features out of the box.
```

> We do our best to make all components screen reader-friendly and offer suggestions for optimizing accessibility throughout our documentation.

Reviewer: ```suggestion
We strive to make all components accessible to end users who require assistive technology, and offer suggestions for optimizing accessibility throughout our documentation.
```

> One example of this is the [Input](/joy-ui/react-input/) component that needs to have a descriptive label linked to help users understand its purpose.

Reviewer: Since this is about the general concept of an input.

```suggestion
For example, an input component needs to have a descriptive label to help users understand its purpose.
```

> Joy UI's Form Control component automatically generates a unique ID that links the Input with the Form Label and Form Helper Text components to ensure you adhere to this principle.

Reviewer: ```suggestion
Joy UI's Form Control component automatically generates a unique ID that links the Input that it wraps with the Form Label and Form Helper Text components, ensuring that your app complies with this guideline:
```

{{"demo": "../../components/list/ExampleCollapsibleList.js"}}

### Developer experience

Joy UI draws its name from the idea that it should spark joy in the creative process of building apps.
Providing an unrivaled developer experience is at the heart of this.

For example, observe how each element of the [Input](/joy-ui/react-input/) component automatically calculates and adjusts its own dimensions relative to the border radius, saving you from the tedium of doing it yourself:

{{"demo": "../../main-features/automatic-adjustment/InputVariables.js"}}

### Accessibility

Joy UI components are built on top of [Base UI's unstyled components and low-level hooks](/base-ui/getting-started/), giving you many accessibility features out of the box.
We strive to make all components accessible to end users who require assistive technology, and offer suggestions for optimizing accessibility throughout our documentation.

Joy UI's Form Control component automatically generates a unique ID that links the Input that it wraps with the Form Label and Form Helper Text components, ensuring that your app complies with this guideline:

{{"demo": "../../components/input/InputField.js"}}

## Start now

> Quickly get started with Joy UI through some of these useful resources:

Reviewer: ```suggestion
Get started with Joy UI today through some of these useful resources:
```
